tspng/extraction.py

```diff
@@ -110,3 +112,34 @@ def extract_from_files(paths: List[str], mime_type: str=MIME_TYPE) -> Dict:
     for path in paths:
         nested_dict[path] = extract_from_file(path, mime_type)
     return nested_dict
+
+def extract_from_folder(path: str, mime_type: str=MIME_TYPE) -> Dict:
+    '''
+    Returns a nested dictionary of metadata from a folder of TSPNG file paths.
+
+        Parameters:
+                path (list[str]): List of file paths
+                mime_type (str): Optional; Media type of file,
+                    default is 'application/vnd.theiascope.io+json'
+
+        Returns:
+                (dict): Dictionary containing metadata of each file
+
+        Raises:
+                Exception: If path is not a directory
+                Exception: If path does not contain a PNG file
+    '''
+    #check if directory
+    if not os.path.isdir(path):
+        raise Exception(f"{path} is not to a directory.")
```

Comment:
I would add a "The" to the error message, i.e., "The {path} is not to a directory.".

Reply:
Change made as requested